### PR TITLE
Add missing overrides in unit tests

### PR DESCRIPTION
**Pull request by 4tXJ7f**
Note: There are still warnings because of the code generated by CxxTest but at least less. Also, I did not clang-format the changes for the review because it changes a lot of formatting.

### CODE REVIEW

File: test/unit/util/rational_white.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ class RationalWhite : public CxxTest::TestSuite {
 
   void testCompareAgainstZero(){
     Rational q(0);
-    TS_ASSERT_THROWS_NOTHING(q == 0;);
+    TS_ASSERT_THROWS_NOTHING(q = 0;);
```
Comment (aniemetz) [inline test/unit/util/rational_white.h]: Why did you change this?

Comment (4tXJ7f) [reply test/unit/util/rational_white.h]: Huh, good catch. Not sure how that happened.
